src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1055 @@
+        EndpointStatistics endpointStatistics = new EndpointStatistics();
+
+        controlOperationsLock.lock();
+        try {
+            if (isValidControlCoolDown() && localStatistics != null && localEndpointStatistics != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Device is occupied. Skipping statistics refresh call.");
+                }
+                extendedStatistics.setStatistics(localStatistics.getStatistics());
+                extendedStatistics.setControllableProperties(localStatistics.getControllableProperties());
+
+                endpointStatistics.setInCall(localEndpointStatistics.isInCall());
+                endpointStatistics.setCallStats(localEndpointStatistics.getCallStats());
+                endpointStatistics.setVideoChannelStats(localEndpointStatistics.getVideoChannelStats());
+                endpointStatistics.setAudioChannelStats(localEndpointStatistics.getAudioChannelStats());
+
+                return Arrays.asList(extendedStatistics, endpointStatistics);
+            }
+
+            Map<String, String> statistics = new HashMap<>();
+            List<AdvancedControllableProperty> controls = new ArrayList<>();
+            retrieveSystemStatus(statistics);
+            retrieveSystemInfo(statistics);
+            retrieveApplications(statistics);
+            retrieveSessions(statistics);
+            retrieveMicrophonesStatistics(statistics);
+            retrieveContentStatus(statistics);
+            retrieveConferencingCapabilities(statistics);
+            retrieveAudioStatus(statistics);
+            retrieveCollaborationStatus(statistics);
+
+            statistics.put("Audio Volume", "");
+            controls.add(createSlider("Audio Volume", 0.0f, 100.0f, Float.valueOf(retrieveVolumeLevel())));
+            statistics.put("Mute Microphones", "");
+            controls.add(createSwitch("Mute Microphones", "On", "Off", retrieveMuteStatus().equals(MuteStatus.Muted)));
+            statistics.put("Mute Local Video", "");
+            controls.add(createSwitch("Mute Local Video", "On", "Off", retrieveVideoMuteStatus()));
+            statistics.put("Reboot", "");
+            controls.add(createButton("Reboot", "Reboot", "Rebooting...", REBOOT_GRACE_PERIOD_MS));
+
+            extendedStatistics.setStatistics(statistics);
+            extendedStatistics.setControllableProperties(controls);
+
+            Integer conferenceId = retrieveActiveConferenceCallStatistics(statistics);
+            boolean validConferenceId = conferenceId != null && conferenceId > -1;
+
+            endpointStatistics.setInCall(validConferenceId);
+            if (validConferenceId) {
+                CallStats callStats = new CallStats();
+                callStats.setCallId(String.valueOf(conferenceId));
+                callStats.setProtocol(statistics.get("Active Conference#Protocol"));
+                callStats.setRequestedCallRate(CALL_RATE);

```

Comment:
This is wrong, if you look at your dial implementation, it uses call rate from the parameter first and only after it is not there it falls on static variable. But you also do not know whether call initiated from adapter or not, so you cannot even use the one from dial (unless you keep track reliably of who initiated the call).

Reply:
Agree, so what should be the best approach here? 
Should it just be just left blank, and then the rest of the information provided (audio/video/content channel stats) would be enough?

Reply:
The rule we use during monitoring is simple. Whatever values are not provided via APIs or cannot be calculated from values provided by APIs should not be set in the statistics (remained with null value).